reader.go

```diff
@@ -23,14 +23,27 @@ type CBORReader struct {
 	pushback     byte
 	pushed       bool
 	messageLimit uint64 // Implement a maximum message size.
+	regTags      map[uint64]reflect.Type
```

Comment:
It would be better to use a `CBORTag` instead of `uint64` in the map key here, then the casts later on would not be necessary.

Reply:
Done